tests/functional/v2/cache/testRedisBackend.js

```diff
@@ -0,0 +1,90 @@
+const assert = require('assert');
+const uuid = require('uuid');
+const { RedisCache } = require('../../../../libV2/cache').backends;
+const { shardFromTimestamp } = require('../../../../libV2/utils');
+const schema = require('../../../../libV2/cache/schema');
+const { generateFakeEvents } = require('../../../utils/v2Data');
+
+const startTime = new Date().getTime();
+const endTime = startTime + 30000; // Add 30 seconds;
+const testValues = generateFakeEvents(startTime, endTime, 50);
+
+describe('Test cache redis backend', () => {
+    let cache;
+    let prefix;
+    beforeEach(done => {
+        prefix = uuid.v4();
+        cache = new RedisCache({}, prefix);
+        cache.connect().then(() => done());
+    });
+
+    it('should create a new redis backend instance', () => {
+        assert.equal(cache instanceof RedisCache, true);
+    });
+
+    testValues.map(event =>
+        it('should set and get a key', () => {
+            const key = schema.getUtapiMetricKey('test', event);
+            return cache
+                .setKey(key, JSON.stringify(event.getValue()))
+                .then(res => {
+                    assert.strictEqual(true, res);
+                    return cache.getKey(key);
+                })
+                .then(value => {
+                    assert.deepStrictEqual(JSON.parse(value), event.getValue());
+                });
+        }));
+
+    testValues.map(event =>
+        it('should add a key to a shard', () => {
+            const shard = shardFromTimestamp(event.timestamp);
+            return cache.addToShard(shard, event).then(res => {
+                assert.strictEqual(true, res);
```

Comment:
not a huge deal but strict equal has parameters of (value, expected) so when reading a failed test log it would say something like. "expected true to equal something" when it might make more sense the other way around. It doesn't really matter much when comparing a bool though.